cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (C) 2014 Harm Weites
+# Copyright (C) 2019 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _select_hostname(self, hostname, fqdn):
+        if fqdn:
+            return fqdn
+        return hostname
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(filename='/etc/rc.conf')
+        return ('/etc/rc.conf', sys_hostname)
+
+    def _read_hostname(self, filename, default=None):
+        return bsd_util.get_rc_config_value('hostname')
+
+    def _write_hostname(self, hostname, filename):
+        bsd_util.set_rc_config_value('hostname', hostname,
+                                     fn='/etc/rc.conf')
+
+    def create_group(self, name, members):
+        group_add_cmd = ['pw', '-n', name]
+        if util.is_group(name):
+            LOG.warning("Skipping creation of existing group '%s'", name)
+        else:
+            try:
+                util.subp(group_add_cmd)
+                LOG.info("Created new group %s", name)
+            except Exception as e:
+                util.logexc(LOG, "Failed to create group %s", name)
+                raise e
+
+        if len(members) > 0:
```

Comment:
consider turning this if on the head, and returning, so we get a little bit more space on for the code below

Reply:
Done, and I've done the same clean up in this patch: https://github.com/canonical/cloud-init/pull/146